### PR TITLE
feat: Test with multiple Keycloak versions

### PR DESCRIPTION
**Pull request by tobiasge**
This extends the test matrix to use multiple Keycloak versions.

### CODE REVIEW

File: .github/workflows/daily.yaml [hunk 1]
```diff
@@ -11,6 +11,9 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        keyclaok-version: ["20.0", "21.0"]
```
Comment (nuwang) [inline .github/workflows/daily.yaml]:
```suggestion
        keycloak-version: ["20.0", "21.0"]
```

File: .github/workflows/daily.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ jobs:
...
+    env:
+      KEYCLOAK_DOCKER_IMAGE_TAG: ${{ matrix.keyclaok-version }}
```
Comment (nuwang) [inline .github/workflows/daily.yaml]:
```suggestion
      KEYCLOAK_DOCKER_IMAGE_TAG: ${{ matrix.keycloak-version }}
```

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -56,9 +56,12 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        keyclaok-version: ["20.0", "21.0"]
     needs:
       - check-commits
       - check-linting
+    env:
+      KEYCLOAK_DOCKER_IMAGE_TAG: ${{ matrix.keyclaok-version }}
```
Comment (nuwang) [inline .github/workflows/lint.yaml]:
```suggestion
      KEYCLOAK_DOCKER_IMAGE_TAG: ${{ matrix.keycloak-version }}
```

File: .github/workflows/lint.yaml [hunk 4]
```diff
@@ -56,9 +56,12 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        keyclaok-version: ["20.0", "21.0"]
```
Comment (nuwang) [inline .github/workflows/lint.yaml]:
```suggestion
        keycloak-version: ["20.0", "21.0"]
```

File: test_keycloak_init.sh [hunk 5]
```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env bash
 
 CMD_ARGS=$1
-KEYCLOAK_DOCKER_IMAGE="quay.io/keycloak/keycloak:latest"
+KEYCLOAK_DOCKER_IMAGE_TAG="${KEYCLOAK_DOCKER_IMAGE_TAG-latest}"
```
Comment (nuwang) [inline test_keycloak_init.sh]:
```suggestion
KEYCLOAK_DOCKER_IMAGE_TAG="${KEYCLOAK_DOCKER_IMAGE_TAG:-latest}"
```

**Review by nuwang — COMMENTED** [review]
This PR is very helpful, because I had my PR break too because of the 21.0 compatibility issue.

**Comment by ryshoooo** [comment]
Hi @tobiasge 

Thanks a lot for the PR :) LGTM I'm just going to ask to add "latest" into the matrix as well, such that we don't lose the checks against the current Keycloak version :)

**Pull request merged**